### PR TITLE
Add public profile page

### PR DESCRIPTION
**Pull request by derricklee43**
https://user-images.githubusercontent.com/39753553/196330280-79414054-3252-4ad2-840f-048b2b97ee2e.mov

### CODE REVIEW

File: src/pages/profile-page/public-profile-page/public-profile-page.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    fetchUserAndRefresh();
+  }, [username]);
+
+  if (publicUser === undefined) {
+    return <LoadingPage />;
+  }
+
+  const decks = publicUser.publicDecks;
+  return (
+    <div className="pg-profile-page">
```
Comment (derricklee43) [inline src/pages/profile-page/public-profile-page/public-profile-page.tsx:29]: TODO: fix naming on this

File: src/pages/search-page/search-page.scss [hunk 2]
```diff
@@ -14,7 +14,11 @@
 }
 
 .search-page-public-deck-tiles {
-  @include wrapping-grid(350px, 32px, 20px);
+  @include deck-tile-wrapping-grid;
+
+  @media screen and (max-width: $medium-screen) {
```
Comment (derricklee43) [inline src/pages/search-page/search-page.scss:19]: TODO: wrong page. Add this selector to `public-profile-page.tsx`

File: src/components/user-details/user-details.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { ProfilePicture } from '@/components/profile-picture/profile-picture';
+import './user-details.scss';
+
+interface UserDetailsProps {
+  username: string;
+  email?: string;
+  yearJoined: number;
```
Comment (jiejasonliu) [inline src/components/user-details/user-details.tsx:8]: maybe this should be `dateJoined` and be a Date? we could display it using `getFormattedDate`
i feel like year is too little info and we're gonna parse out a Date from the response eventually anyway

File: src/models/deck.ts [hunk 4]
```diff
@@ -21,6 +23,36 @@ export interface Deck {
...
 }
 
+export function deckToJson(deck: Deck) {
+  return {
+    title: deck.metaData.title,
+    description: deck.metaData.desc,
+    access: 'Public',
+    default_flang: deck.metaData.frontLang,
+    default_blang: deck.metaData.backLang,
+    userId: 'ad4c76a0-8e0a-4518-b055-5d1dc3ebc4f0', // Todo: replace with id/token
```
Comment (jiejasonliu) [inline src/models/deck.ts:33]: this is legacy and can be removed, the backend knows the user since each request is authorized

**Review by jiejasonliu — APPROVED** [review]
LGTM!

File: src/components/user-details/user-details.tsx (hunk 3 above)
Comment (derricklee43) [reply src/components/user-details/user-details.tsx]: Yeah, that is a good idea. It's fixed now

**Comment by derricklee43** [comment]
I went ahead and fixed the public decks and public users endpoints

**Comment by jiejasonliu** [comment]
nice! everything still looks good to me 😄

**Pull request merged**